core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1872,6 +1872,46 @@ class WorkspaceService(protected val userInfo: UserInfo,
     }
   }
 
+  def adminListWorkspacesWithInvalidEntityCaches: Future[PerRequestMessage] = {
+    asFCAdmin {
+      for {
+        workspacesAndErrors <- dataSource.inTransaction { dataAccess =>
+          for {
+            workspaceIdsAndErrors <- dataAccess.entityCacheQuery.listInvalidCaches
+            workspaces <- dataAccess.workspaceQuery.listByIds(workspaceIdsAndErrors.map(_._1), Option(WorkspaceAttributeSpecs(all=false)))
+          } yield {
+            val errorMap = workspaceIdsAndErrors.toMap
+            workspaces.map { w =>
+              (w, errorMap.get(w.workspaceIdAsUUID).flatten.getOrElse(""))
+            }
+          }
+        }
+        results <- Future.traverse(workspacesAndErrors) { workspaceAndError =>
+          val workspace = workspaceAndError._1
+          loadResourceAuthDomain(SamResourceTypeNames.workspace, workspace.workspaceId, userInfo).map(WorkspaceDetails(workspace, _)).map { details =>
+            WorkspaceDetailsAndCacheError(details, workspaceAndError._2)
+          }
```

Comment:
For my own edification why are we calling sam here?  It seems like we have all the information that we'd need in Rawls' database.

Reply:
This is to populate the [auth domain](https://support.terra.bio/hc/en-us/articles/360026775691-Managing-data-privacy-and-access-with-Authorization-Domains) info for the workspace, so that when listing the workspace, it shows the auth domain. This isn't purely necessary … the result payload for this API is currently only for human consumption, so we can omit anything we want. That said, since this is an admin API I didn't care about the extra call, and if any of these workspaces DO have an auth domain, I thought it would be helpful to display it instead of a misleading empty `"authorizationDomain": []` just because we didn't populate it